tools/shareable-examples/src/utils/get-example-module.ts

```diff
@@ -0,0 +1,163 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { readdirSync } from 'fs';
+import { basename, join, resolve } from 'path';
+import { tsCreateSourceFile } from 'tools/shared/src/utils';
+import {
+  createPrinter,
+  isImportDeclaration,
+  isPropertyAssignment,
+  isVariableStatement,
+  Node,
+  SourceFile,
+  transform,
+  TransformationContext,
+  TransformerFactory,
+  visitEachChild,
+  visitNode,
+  createIdentifier,
+  PropertyAssignment,
+  ImportDeclaration,
+  isSpreadElement,
+  createPropertyAssignment,
+} from 'typescript';
+import { ExampleAstFile } from './examples.interface';
+
+/**
+ * Creates a transformer to prepare the examples module, this includes the following:
+ * - remove all imports from the module that are relative, except the currently processed example
+ * - remove the DT_COMPONENT_EXAMPLES array declaration
+ * - remove the `entryComponents` from the module decorator
+ * - replace the `declarations` in the module decorator with the current example.
+ */
+export function createTransformer(
+  exampleRoot: string,
+  exampleClassName: string,
+): TransformerFactory<SourceFile> {
+  const transformer: TransformerFactory<SourceFile> = (
+    context: TransformationContext,
+  ) => {
+    function visitSpraedDeclarationAndReplaceIt(node: Node): Node | undefined {
+      if (isSpreadElement(node)) {
+        return createIdentifier(exampleClassName);
+      }
+      return visitEachChild(node, visitSpraedDeclarationAndReplaceIt, context);
+    }
+
+    /** Visit a propertyAssignment and determine which assignments need to be rewritten. */
+    function visitPropertyAssignment(
+      node: PropertyAssignment,
+    ): Node | undefined {
+      const assignmentName = node.name.getText();
+      // Replace the spread in the declarations with the one component
+      // we want to keep in there.
+      if (assignmentName === 'declarations') {
+        return visitEachChild(
+          node,
+          visitSpraedDeclarationAndReplaceIt,
+          context,
+        );
+      }
+      // Replace the entryComponents assignment with an exports,
+      // to be able to use the component outside of the declaring module.
+      if (assignmentName === 'entryComponents') {
+        const entryComponentsNode = visitEachChild(
+          node,
+          visitSpraedDeclarationAndReplaceIt,
+          context,
+        );
+        return createPropertyAssignment(
+          'exports',
+          entryComponentsNode.initializer,
+        );
+      }
+      return node;
+    }
+
+    /** Visit an importDeclaration and remove imports if necessary. */
+    function visitImportDeclaration(node: ImportDeclaration): Node | undefined {
+      const importFrom = node.moduleSpecifier.getText().replace(/'/g, '');
```

Comment:
use the `text` property on the module specifier than you don't need to replace something.

```typescript
if (
  !ts.isImportDeclaration(declaration) ||
  !ts.isStringLiteralLike(declaration.moduleSpecifier)
) {
  return;
}
const importFrom = node.moduleSpecifier.text;
```